ingest-app/src/cmr/ingest/validation/generic_document_validation.clj

```diff
@@ -0,0 +1,166 @@
+(ns cmr.ingest.validation.generic-document-validation
+  "Provides functions to validate the ingest generic document"
+  (:require
+   [cheshire.core :as json]
+   [clojure.string :as string]
+   [cmr.common.cache :as cache]
+   [cmr.common.cache.in-memory-cache :as mem-cache]
+   [cmr.common.concepts :as concepts]
+   [cmr.common.generics :as generics]
+   [cmr.common.log :refer (info error)]
+   [cmr.common.services.errors :as errors]
+   [cmr.transmit.metadata-db :as mdb]))
+
+(def schema-validation-cache-key
+  "The cache key for the schema validation functions cache."
+  :schema-validation-functions)
+
+(def SCHEMA_CACHE_TIME
+  "The number of milliseconds schema validation functions will be cached."
+  (* 24 60 60 1000)) ;; 24 hours
```

Comment:
why not forever?

Reply:
If we change the validation at some point, perhaps it could get weird. I mostly jammed the cache in to prevent file IO for every ingest, but this setting is arbitrary.  I'm open to suggestions otherwise but I'd like to keep a ttl.